app/models/manageiq/providers/vmware/infra_manager/vm/reconfigure.rb

```diff
@@ -239,6 +253,32 @@ def add_disk_config_spec(vmcs, options)
     end
   end
 
+  def add_network_adapter_config_spec(vmcs, options)
+    add_device_config_spec(vmcs, VirtualDeviceConfigSpecOperation::Add) do |vdcs|
+      vdcs.device = VimHash.new("VirtualVmxnet3") do |dev|
+        dev.key = rand(-9999..-100) # negative integer as temporary key
+        dev.unitNumber = 0
+        dev.addressType = "Generated"
+        dev.wakeOnLanEnabled = "true"
+        dev.connectable = VimHash.new("VirtualDeviceConnectInfo") do |con|
+          con.allowGuestControl = "true"
+          con.connected = "true"
+          con.startConnected = "true"
+        end
+        dev.backing = VimHash.new("VirtualEthernetCardDistributedVirtualPortBackingInfo") do |bck|
```

Comment:
You need to check the Lan that has been selected to see if it is a distributed or standard network.  We store that as the `:shared` property on the switch so once you find the lan you can check `lan.switch.shared`

Reply:
Added the check for lan.switched.shared.